huber/mock.py

```diff
@@ -0,0 +1,103 @@
+"""Mock interface to a Huber bath."""
+
+import asyncio
+from random import random, choice
+from unittest.mock import MagicMock
+
+from huber.driver import Bath as realBath
+
+
+class Bath(MagicMock):
+    """Mock interface to a Huber bath."""
+
+    def __init__(self, *args, **kwargs):
+        """Init fixed variables."""
+        super().__init__(spec=realBath)
+        self.temp_setpoint = 50
+        self.pump_setpoint = 500
+        self.on = False
+
+    async def __aenter__(self, *args):
+        """Set up connection."""
+        return self
+
+    async def __aexit__(self, *args):
+        """Close connection."""
+        pass
+
+    async def get(self, *args, **kwargs):
+        """Return data structure randomly populated."""
+        await asyncio.sleep(random() * 0.25)
+        return {
+            'on': self.on,  # Temperature control (+pump) active
+            'temperature': {
+                'internal': 23.49,  # Internal temperature, °C
+                'setpoint': self.temp_setpoint  # Temperature setpoint, °C
+            },
+            'pump': {
+                'pressure': random() * 1000,  # Pump head pressure, mbar
+                'speed': random() * 1000,  # Pump speed, rpm
+                'setpoint': self.pump_setpoint  # Pump speed setpoint, rpm
+            },
+            'status': {
+                'circulating': choice([False, True]),  # True if device is circulating
+                'controlling': choice([False, True]),  # True if temperature control is active
+                'error': False,  # True if an uncleared error is present
+                'pumping': choice([False, True]),  # True if pump is on
+                'warning': False,  # True if an uncleared warning is present
+            },
+            'fill': random(),  # Oil level, [0, 1]
+            'maintenance': random()*365,  # Time until maintenance alarm, days
+        }
+
+    async def start(self):
+        """Start the controller and pump."""
+        self.on = True
```

Comment:
I'd add a ~1-3s sleep here. My experience is the Huber sometimes freezes up when the start button is pressed.

This could actually be the Polyscience bath I'm thinking of and not the huber.

Reply:
Add or nah?

Reply:
Do a random(1) async sleep. The longer sleep was the poly science bath but this still can hang occasionally.